src/Angor/Shared/Protocol/FounderTransactionActions.cs

```diff
@@ -152,21 +186,96 @@ public TransactionInfo SpendFounderStage(ProjectInfo projectInfo, IEnumerable<st
         }
 
         _logger.LogInformation($"signed spendingTransaction hex {spendingTransaction.ToHex()}");
-        
+
         var finalTrx = network.CreateTransaction(spendingTransaction.ToHex());
 
-        return new TransactionInfo {Transaction = finalTrx, TransactionFee = totalFee};
+        return new TransactionInfo { Transaction = finalTrx, TransactionFee = totalFee };
+    }
+
+    /// <summary>
+    /// Finds the latest (most recent) expired stage release date from all investment transactions.
+    /// This is needed for Fund/Subscribe projects where different investors may have different stage schedules.
+    /// The founder can only spend a stage after ALL investors' stages have expired.
+    /// </summary>
+    private DateTime FindLatestExpiredStageReleaseDate(ProjectInfo projectInfo, List<string> investmentTransactionsHex, List<int> stageNumbers)
+    {
+        var network = _networkConfiguration.GetNetwork();
+        var latestReleaseDate = DateTime.MinValue;
+
+        for (int i = 0; i < investmentTransactionsHex.Count; i++)
+        {
+            var investmentTrxHex = investmentTransactionsHex[i];
+            var stageNumber = stageNumbers[i];
```

Comment:
So for the last investment transaction we use the last stage and for the first investment transaction we use the first stage index?